lib/app.rb

```diff
@@ -0,0 +1,124 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'teacher'
+require_relative 'class_room'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'person_creator'
+require_relative 'book_creator'
+require_relative 'rental_creator'
+
+# The main application class for managing rentals of books to people.
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    puts 'All books:'
+    if @books.empty?
+      puts 'No books available.'
+    else
+      @books.each do |book|
+        puts "Title:\"#{book.title}\", Author:#{book.author}"
+      end
+    end
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'No people available.'
+      return
+    end
+
+    puts 'All people:'
+    @people.each do |person|
+      puts "[#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a Student (1) or Teacher (2)? [Input the number]:'
+    choice = gets.chomp.to_i
+    person_creator = PersonCreator.new(@people)
+    case choice
+    when 1
+      person_creator.create_student
+    when 2
+      person_creator.create_teacher
+    end
+  end
+
+  def create_book
+    book_creator = BookCreator.new(@books)
+    book_creator.create_book
+  end
+
+  def create_rental
+    rental_creator = RentalCreator.new(@books, @people, @rentals)
+    rental_creator.create_rental
+  end
+
+  # def list_rentals_by_person_id
+  #   print 'Person id: '
+  #   person_id = gets.chomp.to_i
+  #   person = @people.find { |p| p.id == person_id }
+
+  #   unless person
+  #     puts "Could not find person with id #{person_id}."
+  #     return
+  #   end
+
+  #   rentals = @rentals.select { |r| r.person == person }
+
+  #   if rentals.empty?
+  #     puts "#{person.name} (id: #{person.id}) has no rentals."
+  #   else
+  #     puts "All rentals for #{person.name} (id: #{person.id}):"
+  #     rentals.each do |rental|
+  #       puts "Date: #{rental.date}, Book \"#{rental.book.title}\" by #{rental.book.author}"
+  #     end
+  #   end
+  # end
```

Comment:
- [ Required ] I think it would be better if you remove any piece of commented code from your code base, commented code makes your project unprofessional and is usually considered a sign of code that is not clean. Kindly rectify this.

Reply:
Thank you @Felix45